test/bin/scenario_container.sh

```diff
@@ -0,0 +1,162 @@
+set -xeuo pipefail
+
+cat /proc/sys/kernel/random/uuid > /proc/sys/kernel/random/boot_id
+
+if [ -e /var/container-init-ok ]; then
+    echo "Container already initialized"
+    exit 0
+fi
+
+EOF
+    chmod +x "${output_dir}/init.sh"
+    process_kickstart_into_script "${name}"
+
+    cat << EOF >> "${output_dir}/init.sh"
+touch /var/container-init-ok
+
+EOF
+
+    cat << 'EOF' > "${output_dir}/entrypoint.sh"
+#!/bin/bash
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+bash -x "${SCRIPTDIR}/init.sh" &> /var/log/container_init.log
+exec /sbin/init
+EOF
+    chmod +x "${output_dir}/entrypoint.sh"
+}
+
+launch_container() {
+    local name="host1"
+    local image=""
+    local memory=4096
+    local vcpus=2
+
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --name|--image|--vcpus|--memory)
+                var="${1/--/}"
+                if [ -n "$2" ] && [ "${2:0:1}" != "-" ]; then
+                    declare "${var}=$2"
+                    shift 2
+                else
+                    error "Failed parsing arguments: ${var} value not set"
+                    record_junit "${name}" "ctr-launch-args" "FAILED"
+                    exit 1
+                fi
+                ;;
+            *)
+                error "Invalid argument: ${1}"
+                record_junit "${name}" "ctr-launch-args" "FAILED"
+                exit 1
+                ;;
+        esac
+    done
+
+    if [ -z "${image}" ]; then
+        echo "--image is empty"
+        exit 1
+    fi
+
+    local -r full_image="${MIRROR_REGISTRY_URL}/${image}"
+
+    record_junit "${name}" "ctr-launch-args" "OK"
+
+    local -r full_ctr_name=$(full_vm_name "${name}")
+
+    if sudo podman inspect -f "{{.Id}}" "${full_ctr_name}" &> /dev/null; then
+        echo "Container '${full_ctr_name}' already exists."
+        record_junit "${name}" "launch_ctr" "SKIP"
+        exit 0
+    fi
+
+    sudo modprobe openvswitch
+
+    prepare_kickstart "${name}" kickstart-bootc-container.ks.template ""
+    create_container_entrypoint "${name}"
+    local -r output_dir="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${name}/"
+    touch "${output_dir}/boot_id"
+
+    if ! sudo podman run \
+        --detach \
+        --privileged \
+        --restart always \
+        --hostname "${full_ctr_name}" \
+        --volume /dev:/dev \
+        --volume "${output_dir}/boot_id":/proc/sys/kernel/random/boot_id \
+        --volume /var/lib/containers/storage:/var/lib/containers/storage \
+        --volume "${output_dir}":/tmp/ctr_script \
+        --entrypoint '/tmp/ctr_script/entrypoint.sh' \
+        --memory "${memory}m" \
+        --cpus "${vcpus}" \
+        --name "${full_ctr_name}" \
+        --tls-verify=false \
+        "${image}"; then
+        record_junit "${name}" "launch_ctr" "FAILED"
+        exit 1
+    fi
+
+    local -r ip=$(sudo podman inspect "${full_ctr_name}" | jq -r '.[0].NetworkSettings.IPAddress')
+    echo "Container ${full_ctr_name} has IP ${ip}"
```

Comment:
Is there not a race here with the container startup in the detached mode?

Reply:
Haven't had a problem honestly. I think `-d` is different from e.g. systemd's --no-block - the container is created and started, it's just isn't attached to.
But we can poll for IP first for safety

Reply:
Yeah, let's poll. It may be necessary when running tests in parallel on an overloaded system.